src/schema/validator.js

```diff
@@ -542,6 +542,67 @@ export class SchemaValidator {
       ),
       errors: true,
     });
+
+    const getPrivilegedPermissionsSet =
+      this.getPrivilegedPermissionsSet.bind(this);
+
+    validator.addKeyword({
+      keyword: 'postprocess',
+      validate: function validate(keywordSchemaValue, propValue) {
+        if (keywordSchemaValue === 'checkPrivilegePermission') {
+          const privilegedPermissions = getPrivilegedPermissionsSet(validator);
+          const found = new Set();
+
+          for (const permission of propValue) {
+            if (privilegedPermissions.has(permission)) {
+              found.add(permission);
+            }
+          }
+
+          if (found.size > 0) {
+            validate.errors = [
+              {
+                keyword: 'checkPrivilegedPermission',
+                params: { privilegedPermissions: Array.from(found) },
+              },
+            ];
+            return false;
+          }
+        } else if (keywordSchemaValue === 'canUseExperiments') {
+          validate.errors = [{ keyword: 'canUseExperiments' }];
+          return false;
+        } else if (keywordSchemaValue === 'requirePrivilege') {
+          validate.errors = [{ keyword: 'requirePrivilege' }];
+          return false;
+        }
+
+        return true;
+      },
+    });
+  }
+
+  getPrivilegedPermissionsSet(validator) {
+    const schemaManifest = validator.getSchema('manifest').schema;
+    const schemaPermissionPrivileged =
+      schemaManifest.types.PermissionPrivileged.anyOf;
+
+    const results = new Set();
+    for (const schemaPermission of schemaPermissionPrivileged) {
+      const { type, $ref } = schemaPermission;
+      if (type === 'string' && schemaPermission.enum) {
+        schemaPermission.enum.forEach((str) => results.add(str));
+      } else if ($ref) {
+        const [namespace] = $ref.split('#');
+        const schemaNamespace = validator.getSchema(namespace).schema;
+        schemaNamespace.permissions.forEach((str) => results.add(str));
+      }
+    }
+    if (results.size === 0) {
+      throw new Error(
+        'Unable to retrieve the Privileged Permissions set from the schema data'
+      );
+    }
+    return results;
```

Comment:
This looks a bit packed but that also looks good to me.

Reply:
yeah, it is

Ideally this would be much simpler if the schema imported becomes simpler ("simpler" as in "using less indirections", due to having to resolve $extend, $ref etc) or if we would be simplify the schema with those normalizations that I was looking into and mentioned to you some times ago.

Unfortunately I didn't got time to complete that and it was risky enough to do that along with updating to ajv v8, I feel that changing the format during the import is something that may be safer if done in its own separate PR (separately from other changes introducing new rules, or updating a complex-enough dependency like ajv).